src/http.c

```diff
@@ -779,10 +779,45 @@ static void free_http(tlsuv_http_t *clt) {
     tlsuv_tls_link_free(&clt->tls_link);
 }
 
+
 int tlsuv_parse_url(struct tlsuv_url_s *url, const char *urlstr) {
     memset(url, 0, sizeof(struct tlsuv_url_s));
 
     const char *p = urlstr;
+
+    int file_prefix_len=strlen("file:/");
+    // special handling for file:/, file://, file://host/, file:///
+    if (strncmp(urlstr, "file:/", file_prefix_len) == 0) {
+        url->scheme = p;
+        url->scheme_len = 4; // strlen("file")
+        p += file_prefix_len;
+
+        if(p[0] == '/') {
+            p++;
+            if(p[0] == '/') {
+                // file:/// means empty hostname
+                p ++;
+            } else {
+                // file://path means there must be a hostname. find the next slash
+                char *pos = strchr(p, '/');
+                if (pos != NULL) {
+                    int index = pos - p;
+                    url->hostname= p;
+                    url->hostname_len = index;
+                    p += index;
+                } else {
+                    printf("The character '/' was not found in the string.\n");
```

Comment:
remove

Reply:
yeah. i already have that removed. it'll be in the next push. thx